src/components/experiments/HeaderButtons/useOnForkButtonPressed.tsx

```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect } from "react";
+import { api } from "~/utils/api";
+import { useExperiment, useHandledAsyncCallback } from "~/utils/hooks";
+import { signIn, useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import queryString from "query-string";
+
+export const useOnForkButtonPressed = () => {
+  const utils = api.useContext();
+  const router = useRouter();
+
+  const user = useSession().data;
+  const experiment = useExperiment();
+
+  const forkMutation = api.experiments.fork.useMutation();
+
+  const [onFork, isForking] = useHandledAsyncCallback(async () => {
+    if (!experiment.data?.id) return;
+    const forkedExperiment = await forkMutation.mutateAsync({ id: experiment.data.id });
+    await utils.experiments.list.invalidate();
+    await router.push({ pathname: "/experiments/[id]", query: { id: forkedExperiment.id } });
+  }, [forkMutation, experiment.data?.id, router]);
+
+  const onForkButtonPressed = useCallback(() => {
+    if (user === null) {
+      // Parse the current query parameters
+      const currentQueries = queryString.parse(window.location.search);
+
+      // Set the fork parameter to true
+      const newQueries = queryString.stringify({
+        ...currentQueries,
+        fork: "true",
+      });
+
+      signIn("github", {
+        callbackUrl: `${window.location.origin}${window.location.pathname}?${newQueries}`,
```

Comment:
just verifying you've confirmed this works and redirects correctly back to the page, right?

Reply:
Nah, figured I'd let the users test it out for us. Saves engineering time.

Reply:
Lol yes I checked